05-JWT-Basics/preferred/middleware/auth.js

```diff
@@ -0,0 +1,23 @@
+const jwt = require('jsonwebtoken');
+
+const authenticationMiddleware = async (req, res, next) => {
+  const authHeader = req.headers.authorization;
+  if (!authHeader || !authHeader.startsWith('Bearer ')) {
+    return res.status(401).json({
+      message: 'No token provided',
+    });
+  }
+  const token = authHeader.split(' ')[1];
+  try {
+    const decoded = jwt.verify(token, process.env.JWT_SECRET);
```

Comment:
What are we verifying?

Reply:
This middleware function checks if a user is authorized to access protected resources by verifying the authenticity of a JWT token.

Reply:
> verifying the authenticity of a JWT token.

How do we verify the authenticity of a JWT?

Reply:
We use the ```jwt.verify()``` method from the ```jsonwebtoken``` library. This method takes two arguments: the token to be verified and a secret key to decode the signature. The token we extract from ```req.header.authorization```.